microprofile.h

```diff
@@ -192,9 +192,13 @@ typedef uint16_t MicroProfileGroupId;
 
 #include <stdint.h>
 
+#ifdef MICROPROFILE_EXPORT
+#include "microprofile.export.h"

```

Comment:
I dont understand why this is needed (instead of just the macros)?

Reply:
So it can be built as a shared library with CMake (when `BUILD_SHARED_LIBS` is set to true). To export symbols on Windows, `__declspec(dllexport)` is needed (and `__declspec(dllimport)` when importing, but that is mostly optional) so that the final `.dll` file will have all needed symbols exported. I used `generate_export_header` so the export header generation is handled by CMake.

Reply:
Alright. The original intention was that you'd just defined MICROPROFILE_API to whatever you need, but I suppose we can support this one as well.

Thank you for the PR